index.js

```diff
@@ -4368,9 +4620,31 @@ Agent.prototype.request = function (requestMessage, rinfo) {
 					oid: requestPdu.varbinds[i].oid
 				});
 
-				if ( requestPdu.type == PduType.SetRequest ) {
-					mibRequests[i].setType = requestPdu.varbinds[i].type;
-					mibRequests[i].setValue = requestPdu.varbinds[i].value;
+				if ( requestPdu.type == PduType.SetRequest && instanceNode.address.length >= 2 ) {
+					var addrLen = instanceNode.address.length;
+					var row = instanceNode.address[addrLen - 1];
```

Comment:
It's important to note that the row identifier in an OID is not necessarily the single final number in the OID.  There are several scenarios (composite index, non-integer index, etc.) where this isn't the case - the row could be the last several numbers in the OID.  The column, however, is always a single integer from the OID.  For the column, there's already MibNode.getTableColumnFromInstanceNode() to get this.  A similar function could be written to get the row index (would be an array of integers).